src/OAuth/OAuth1/Service/Etsy.php

```diff
@@ -27,12 +52,19 @@ public function __construct(
         }
     }
 
+    public function setScopes($requested_scopes = array())
+    {
+        if ($requested_scopes) {
+            $this->scopes = '?scope=' . implode('%20', $requested_scopes);
+        }
+    }
+
     /**
      * {@inheritdoc}
      */
     public function getRequestTokenEndpoint()
     {
-        return new Uri($this->baseApiUri . 'oauth/request_token');
+        return new Uri($this->baseApiUri . 'oauth/request_token' . $this->scopes);
```

Comment:
There needs to be unit tests for zero, one and multiple scopes set correctly on the Uri.